### PR TITLE
Added voltage magnitude and angle to SecurityAnalysis branch results extension

### PR DESCRIPTION
**Pull request by obrix**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Add extensions to security analysis results to have voltage results associated to Branch and Transformers. 

**What is the current behavior?**

Voltage results are only available in BusResult

**What is the new behavior (if this is a feature change)?**

Voltage results are available in Branch and ThreeWindingsTransformer results.

**Does this PR introduce a breaking change or deprecate an API?**
No

**Other information**:
n/a

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/network/impl/LfBranchImpl.java [hunk 1]
```diff
@@ -165,7 +165,11 @@ public BranchResult createBranchResult(double preContingencyBranchP1, double pre
         var branchResult = new BranchResult(getId(), p1.eval() * PerUnit.SB, q1.eval() * PerUnit.SB, currentScale1 * i1.eval(),
                                             p2.eval() * PerUnit.SB, q2.eval() * PerUnit.SB, currentScale2 * i2.eval(), flowTransfer);
         if (createExtension) {
-            branchResult.addExtension(OlfBranchResult.class, new OlfBranchResult(piModel.getR1(), piModel.getContinuousR1()));
+            branchResult.addExtension(OlfBranchResult.class, new OlfBranchResult(piModel.getR1(), piModel.getContinuousR1(),
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/impl/LfBranchImpl.java:168]: You have three issues here:
- Bus can be null yes, but maybe you can find a way to hide all of this in `OlfBranchResult` constructor.
- v is still in per-unit quantities and we don't want that as output.
- angle is still in radians but output in degrees.

File: src/main/java/com/powsybl/openloadflow/network/impl/LfLegBranch.java [hunk 2]
```diff
@@ -194,17 +194,36 @@ public void updateFlows(double p1, double q1, double p2, double q2) {
                 .setQ(q1 * PerUnit.SB);
     }
 
-    public static ThreeWindingsTransformerResult createThreeWindingsTransformerResult(LfNetwork network, String threeWindingsTransformerId) {
+    private double getV() {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/impl/LfLegBranch.java:197]: You don't need these 2 methods, no?

Comment (obrix) [reply src/main/java/com/powsybl/openloadflow/network/impl/LfLegBranch.java:197]: They are only there to factorize the null test.

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/network/impl/LfLegBranch.java:197]: Use the methods of  `AbstractLfImpedantBranch`, no?

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/network/impl/LfLegBranch.java:197]:
````
            result.addExtension(OlfThreeWindingsTransformerResult.class, new OlfThreeWindingsTransformerResult(
                    leg1.getV1() * leg1.legRef.get().getTerminal().getVoltageLevel().getNominalV(),
                    leg2.getV1() * leg2.legRef.get().getTerminal().getVoltageLevel().getNominalV(),
                    leg3.getV1() * leg3.legRef.get().getTerminal().getVoltageLevel().getNominalV(),
                    Math.toDegrees(leg1.getAngle1()),
                    Math.toDegrees(leg2.getAngle1()),
                    Math.toDegrees(leg3.getAngle1())));
````

**Review by annetill — APPROVED** [review]

**Review by jeandemanged — APPROVED** [review]
thanks!
ok to merge, but @obrix please fill in the PR description

File: src/main/java/com/powsybl/openloadflow/network/impl/OlfThreeWindingsTransformerResult.java
Comment (jeandemanged) [inline src/main/java/com/powsybl/openloadflow/network/impl/OlfThreeWindingsTransformerResult.java:1]: copyright & author missing

**Review by jeandemanged — CHANGES_REQUESTED** [review]

**Review by jeandemanged — APPROVED** [review]

**Pull request merged**